### PR TITLE
STDTC-31: add tests  for SearchForm with jest/react-testing-library

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose

https://issues.folio.org/browse/STDTC-31
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by vashjs — APPROVED** [review]

File: lib/SearchForm/SearchForm.test.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    it('should display search form', () => {
+      setupSearchForm({
+        searchTerm,
+        handleChange: handleChangeMock,
+        handleSubmit: handleSubmitMock,
+      });
+
+      expect(document.querySelector('[data-test-search-form]')).toBeVisible();
+    });
```
Comment (NikitaSedyx) [inline lib/SearchForm/SearchForm.test.js:58]: empty line plz

File: lib/SearchForm/SearchForm.test.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    it('should display search form submit button', () => {
+      setupSearchForm({
+        searchTerm,
+        handleChange: handleChangeMock,
+        handleSubmit: handleSubmitMock,
+      });
+
+      expect(screen.getByRole('button', { name: 'stripes-smart-components.search' })).toBeVisible();
```
Comment (NikitaSedyx) [inline lib/SearchForm/SearchForm.test.js:79]: ideally they should be 2 different cases

File: lib/SearchForm/SearchForm.test.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      expect(submitButton).toBeDisabled();
+    });
+
+    it('should display the default search field id', () => {
+      setupSearchForm({});
+
+      const searchBox = screen.getByRole('searchbox');
+
+      expect(searchBox).toHaveAttribute('id', 'input-search-field');
+    });
```
Comment (NikitaSedyx) [inline lib/SearchForm/SearchForm.test.js:135]: you have different codestyle, somewhere with line, somewhere without

File: lib/SearchForm/SearchForm.test.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply lib/SearchForm/SearchForm.test.js:79]: > ideally they should be 2 different cases

Ye, I'm testing enable or disable button in cases below.

Comment (NikitaSedyx) [reply lib/SearchForm/SearchForm.test.js:80]: why do you check that button is visible when you test enable/disable?

Comment (UladzislauKutarkin) [reply lib/SearchForm/SearchForm.test.js:80]: Just check if it present on the page first of all, to be sure in that fact. And after that I check it enable or not. In other cases I can be sure, that button visible on the page, so I can test on enable/disable.

File: lib/SearchForm/SearchForm.test.js (hunk 3 above)
Comment (UladzislauKutarkin) [reply lib/SearchForm/SearchForm.test.js:136]: fixed.

File: lib/SearchForm/SearchForm.test.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply lib/SearchForm/SearchForm.test.js:58]: done in whole file.

**Review by NikitaSedyx — APPROVED** [review]

File: lib/SearchForm/SearchForm.test.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+      });
+
+      const submitButton = screen.getByRole('button', { name: 'stripes-smart-components.search' });
+
+      userEvent.click(submitButton);
+
+      expect(handleSubmitMock).toHaveBeenCalled();
+    });
+
+    it('should display search form submit button disabled when search filed is empty', () => {
```
Comment (AndreiShumski2) [inline lib/SearchForm/SearchForm.test.js:112]: search field

**Review by AndreiShumski2 — APPROVED** [review]

**Pull request merged**